app/test/e2e/query-get.spec.ts

```diff
@@ -0,0 +1,123 @@
+        const query: string = `select * from ${timestamp}`;
+        const response: Record<string, any> = await requester
+            .get(
+                `/api/v1/gfw/query/${timestamp}?sql=${encodeURI(query)}&geostore_id=`,
+            )
+            .send(requestBody);
+
+        ensureCorrectError(
+            response,
+            "This operation is only supported for datasets with connectorType 'rest'",
+            422,
+        );
+    });
+
+    it('Query to dataset without a supported provider should fail', async () => {
+        const timestamp: string = String(new Date().getTime());
+
+        createMockGetDataset(timestamp, { provider: 'foo' });
+
+        const requestBody: Record<string, any> = {};
+        const query: string = `select * from ${timestamp}`;
+        const response: Record<string, any> = await requester
+            .get(`/api/v1/gfw/query/${timestamp}?sql=${query}`)
+            .send(requestBody);
+
+        ensureCorrectError(
+            response,
+            "This operation is only supported for datasets with provider 'gfw'",
+            422,
+        );
+    });
+
+    it('Query without sql or fs parameter should return bad request', async () => {
+        const timestamp: string = String(new Date().getTime());
+
+        createMockGetDataset(timestamp);
+
+        const response: Record<string, any> = await requester
+            .get(`/api/v1/gfw/query/${timestamp}`)
+            .send();
+
+        ensureCorrectError(response, 'sql or fs required', 400);
+    });
+
+    it('Send query should return result(happy case)', async () => {
+        const timestamp: string = String(new Date().getTime());
+        const sql: string = 'SELECT * from DATA LIMIT 2';
+
+        createMockGetDataset(timestamp);
+        createMockSQLQuery(sql);
+        createMockConvertSQL(sql);
+
+        const response: Record<string, any> = await requester
+            .get(`/api/v1/gfw/query/${timestamp}`)
+            .query({ sql })
+            .send();
+
+        response.status.should.equal(200);
+        response.body.should.have.property('data').and.instanceOf(Array);
+        response.body.should.have.property('meta').and.instanceOf(Object);
+
+        const { meta, data } = response.body;
+        data.should.deep.equal(DEFAULT_RESPONSE_SQL_QUERY.data);
+
+        meta.should.have.property('cloneUrl').and.instanceOf(Object);
+        // eslint-disable-next-line camelcase
+        const {
+            cloneUrl: { http_method: httpMethod, url, body },
+        } = meta;
+        httpMethod.should.equal('POST');
+        url.should.equal(`/dataset/${timestamp}/clone`);
+        body.should.have.property('dataset').and.instanceOf(Object);
+
+        const { datasetUrl, application } = body.dataset;
+        application.should.deep.equal(['your', 'apps']);
+        datasetUrl.should.equal(
+            `/query/${timestamp}?sql=${encodeURI(sql).replace('*', '%2A')}`,
```

Comment:
this `.replace()` call feels evil, as there are more characters that encodeURI does not escape. not a big deal, look into a replacement if you feel like it

Reply:
Will keep an eye out of that usage. Leaving it as here as it's being used for fixed known string